R/conda.R

```diff
@@ -1106,9 +1106,26 @@ get_python_conda_info <- function(python) {
     else
       file.path(root, "bin/conda")
   } else {
-    # not base env, parse conda-meta history to find the conda binary
-    # that created it
-    conda <- python_info_condaenv_find(root)
+    # not base env
+    conda <- find_conda()
```

Comment:
Thank you for the PR! I noticed something in the implementation that might need clarification. 

The purpose of `get_python_conda_info()` is to locate the specific `conda` used to create a given `python` conda environment. Since it's common for users to have multiple versions of `conda` installed, this function focuses on identifying the correct one for the environment.

On the other hand, `find_conda()` is meant to find the user's preferred `conda` executable based on expressed preferences in `options()` or other heuristics.

It seems like using `find_conda()` within `get_python_conda_info()` may not be the right approach here. What do you think?

Reply:
I think my specific issue is when it hits 

```
  if(!file.exists(conda))
    conda <- NA
```

so I guess in other locations if `NA` is passed I can change it to use `find_conda` then

Reply:
Oh I guess `conda_binary` already handles that specific case here

https://github.com/rstudio/reticulate/blob/4277e2dfc4da7403028a05dc3db2d9591d22a212/R/conda.R#L531-L539

The only reason my error was occuring was because `get_python_conda_info` used `python_info_condaenv_find` directly which didn't handle invalid values from the history